### ISSUE TITLE
Auto Loader experimental effect should increase Sustained DPS.

### ISSUE THREAD

**Issue by ulyssesric** [EDCD/coriolis#326]
After you apply the Auto Loader experimental effect to Cannon/Multi-Cannon, ammo in clip will be refilled at steady rate even when firing. Which means the time to deplete clip is effectively increased. I can't find player videos after 3.0 but this old video can give you an rough idea:

https://www.youtube.com/watch?v=GZ5iM8BT-hc

This video is taken in older version that Auto Loader will decrease clip size up to 10% (the negative effect is now removed).  In the video we can see the clip size is ~75 shots and the M-size Gimbaled Multi-Cannon with G5 Overcharged can fire 7.7 rounds per seconds. Without Auto Loader the clip should be depleted in less then 10 seconds, but in the video he can continuously firing for 17 seconds before reloading the clip.

The DPS of a M-size Gimbaled Multi-Cannon (2F) with G5 Overcharge / no experimental effect is 21.4, while the EDCD says the SDPS is 14.5. We can understand that SDPS is calculated based on effective time of firing: you have to wait for 5.5 seconds after you deplete the clip. This is what we get:

SDPS = DPS * (Clip_Deplete_Time) / (Clip_Deplete_Time + Reload_Time)

while Clip_Deplete_Time = Clip_Size / RoS

So we have: 14.5 = 21.4 * (90/7.7) / (90/7.7 + 5.5)

It makes sense for normal conditions.

However, with Auto Loader, it is equivalent to calculate the effective time of firing with a lower RoS. That is, when you fire 6 rounds and get 3 rounds refilled, you're effectively doubled the firing time before depleting clip.

From the Nov. 2016 hot fix note (https://www.alpha-orbital.com/news/hotfix-and-balancing-patch-for-engineer-blueprints-announced) we knew that the reload rate of Auto Loader is increased to 1 round per 2 firing interval, which means the **effective RoS is halved**. For the same condition (2F MC + G5 Overcharge), the clip deplete time should be 23.3 seconds with Auto Loader.

So the SDPS with Auto Loader should be:

SDPS = 21.4 * 23.3 / (23.3 + 5.5) = 17.3

Simply put: Auto Loader will not effect DPS but will increase SDPS. And this makes sense.

**Comment by felixlinker** [comment]
Possible idea for a fix: Calculate a *sustained rate of fire* value additional to the *rate of fire*  value. We could use that value then [here](https://github.com/EDCD/coriolis/blob/master/src/app/shipyard/Module.js#L536).

**Comment by felixlinker** [comment]
Okay, figured out that SDPS does not take the distributor into account (maybe something for future times). I have now implemented a fix for that. Thanks for the detailed bug report. Helped me out implementing the fix a lot!

**Comment by willyb321** [comment]
Fixed in c9cd6b175da1ac95589900dc130acefa13c93401
